Providers/RustSyncManager.swift

```diff
@@ -0,0 +1,628 @@
+        }
+    }
+
+    var prefsForSync: Prefs {
+        return self.prefs.branch("sync")
+    }
+
+    public func onAddedAccount() -> Success {
+        // Only sync if we're green lit. This makes sure that we don't sync unverified
+        // accounts.
+        guard self.profile.hasSyncableAccount() else { return succeed() }
+
+        self.beginTimedSyncs()
+        return self.syncEverything(why: .didLogin)
+    }
+
+    public func onRemovedAccount() -> Success {
+        let clearPrefs: () -> Success = {
+            withExtendedLifetime(self) {
+                // Clear prefs after we're done clearing everything else -- just in case
+                // one of them needs the prefs and we race. Clear regardless of success
+                // or failure.
+
+                // This will remove keys from the Keychain if they exist, as well
+                // as wiping the Sync prefs.
+
+                // XXX: `Scratchpad.clearFromPrefs` and `clearAll` were pulled from
+                // `SyncStateMachine.clearStateFromPrefs` to reduce RustSyncManager's
+                // dependence on the swift sync state machine logid. This will make
+                // refactoring or eliminating that code easier once the rust sync manager
+                // rollout is complete.
+                Scratchpad.clearFromPrefs(self.prefsForSync.branch("scratchpad"))
+                self.prefsForSync.clearAll()
+            }
+            return succeed()
+        }
+        syncQueue.async { [unowned self] in
+            self.syncManagerAPI.disconnect()
+        }
+        return clearPrefs()
+    }
+
+    private func getEngineEnablementChangesForAccount() -> [String: Bool] {
+        var engineEnablements: [String: Bool] = [:]
+        // We just created the account, the user went through the Choose What to Sync
+        // screen on FxA.
+        if let declined = UserDefaults.standard.stringArray(
+            forKey: "fxa.cwts.declinedSyncEngines") {
+            declined.forEach { engineEnablements[$0] = false }
+            UserDefaults.standard.removeObject(forKey: "fxa.cwts.declinedSyncEngines")
+        } else {
+            // Bundle in authState the engines the user activated/disabled since the
+            // last sync.
+            RustTogglableEngines.forEach { engine in
+                let stateChangedPref = "engine.\(engine).enabledStateChanged"
+                if self.prefsForSync.boolForKey(stateChangedPref) != nil,
+                   let enabled = self.prefsForSync.boolForKey("engine.\(engine).enabled") {
+                    engineEnablements[engine] = enabled
+                    self.prefsForSync.setObject(nil, forKey: stateChangedPref)
+                }
+            }
+        }
+
+        if !engineEnablements.isEmpty {
+            logger.log("""
+                engines to enable:
+                \(engineEnablements.compactMap { $0.value ? $0.key : nil })
+                """,
```

Comment:
Wondering if this string format will save the log on multiple lines (I've had that issue in the past for other loggers), can we double check to make sure it's saved on one line?

Reply:
I had a look, and turns out yes using `"""` will make the log multiline in our logger too. This could make it harder to understand logs. I would suggest just using standard strings instead with `"`

My little test:
![Screen Shot 2023-03-06 at 5 38 38 PM](https://user-images.githubusercontent.com/11338480/223267219-0ebad250-65d6-482b-9406-f521fd389220.png)

![Screen Shot 2023-03-06 at 5 37 34 PM](https://user-images.githubusercontent.com/11338480/223267226-0f40fd02-ddf9-494c-86cb-148c8a71110c.png)

Reply:
Sounds good! The only reason I used this format was to prevent the lines from being too long. 😊

Reply:
Yeah it's always a battle with strings to try to make them not too long haha. My only way otherwise to make them shorter is using variables like
```
let engines = "\(engineEnablements.compactMap { $0.value ? $0.key : nil })"
logger.log("engines to enable: \(engines)",
[...]
```